### ISSUE TITLE
Unclear error when updating a names_for_user variable in correct_code

### ISSUE THREAD

**Issue by bowmanat** [inducer/relate#319]
If you try to update and then use a variable in correct_code that came from names_for_user, the grading code will raise an error telling you that the variable is unbound.  It is easy to work around once you know what is happening, but if the system does not want the correct_code section to modify these variables, it would at least be helpful to have a more clear error message.

This does not happen when students try to modify the given variables in their code.

The attached file is a simple example of code that breaks.

[bad_code.txt](https://github.com/inducer/relate/files/959184/bad_code.txt)

**Comment by inducer** [comment]
I agree that this isn't great, but it's a consequence of Python's scoping rules. I don't see how we could work around those.

**Comment by inducer** [comment]
Well, that said, we could preface the correct code with 

```
global VAR1, VAR2
```

which would modify those variables in the global scope--but then, that's arguably worse.
